Plugin/NodOSVRPlugin.cpp

```diff
@@ -0,0 +1,180 @@
+    const float sinHalfPitch = (float)sin(pitch / 2.0f);
+    const float cosHalfPitch = (float)cos(pitch / 2.0f);
+    const float sinHalfRoll = (float)sin(roll / 2.0f);
+    const float cosHalfRoll = (float)cos(roll / 2.0f);
+
+    mData[0] = cosHalfRoll * cosHalfPitch * cosHalfYaw
+        + sinHalfRoll * sinHalfPitch * sinHalfYaw;
+    mData[1] = cosHalfRoll * sinHalfPitch * sinHalfYaw
+        + cosHalfPitch * cosHalfYaw * sinHalfRoll;
+    mData[2] = cosHalfRoll * cosHalfPitch * sinHalfYaw
+        - sinHalfRoll * cosHalfYaw * sinHalfPitch;
+    mData[3] = cosHalfYaw*sinHalfPitch*cosHalfRoll -
+        sinHalfYaw*cosHalfPitch*sinHalfRoll;
+}
+
+class Backspin {
+public:
+    Backspin(OSVR_PluginRegContext ctx) {
+        /// Create the initialization options
+        OSVR_DeviceInitOptions opts = osvrDeviceCreateInitOptions(ctx);
+
+        /// config channels
+        osvrDeviceAnalogConfigure(opts, &m_analog, 3);
+        osvrDeviceButtonConfigure(opts, &m_button, 10);
+        osvrDeviceTrackerConfigure(opts, &m_orientation);
+        /// Create the device token with the options
+        m_dev.initAsync(ctx, std::to_string(deviceCount).c_str(), opts);
+        deviceCount++;
+
+        /// Send JSON descriptor
+        m_dev.sendJsonDescriptor(NodOSVR);
+    }
+    osvr::pluginkit::DeviceToken m_dev;
+    OSVR_AnalogDeviceInterface m_analog;
+    OSVR_ButtonDeviceInterface m_button;
+    OSVR_TrackerDeviceInterface m_orientation;
+};
+
+std::vector<Backspin*> devices;
+
+void OpenSpatialEventFired(NodEvent ev)
+{
+    if (ev.type == EventType::ServiceReady)
+    {
+        std::cout << "[NOD PLUGIN] Service Ready" << std::endl;
+        for (int i = 0; i < NodNumRings(); i++)
+        {
+            NodSubscribe(Modality::GameControlMode, NodGetRingName(i));
+            NodSubscribe(Modality::ButtonMode, NodGetRingName(i));
+            NodSubscribe(Modality::EulerMode, NodGetRingName(i));
+        }
+    }
+    if (ev.type == EventType::AnalogData)
+    {
+        Backspin* dev = devices.at(ev.sender);
+        OSVR_AnalogState analogval[3];
+        analogval[0] = ev.trigger;
+        analogval[1] = ev.x;
+        analogval[2] = ev.y;
+        osvrDeviceAnalogSetValues(dev->m_dev, dev->m_analog, analogval, 3);
+    }
+    if (ev.type == EventType::Button)
+    {
+        Backspin* dev = devices.at(ev.sender);
+        int index = ev.buttonID;
+        int val = (ev.buttonState == UP) ? 0 : 1;
+        osvrDeviceButtonSetValue(dev->m_dev, dev->m_button, val, index);
+    }
+    if (ev.type == EventType::EulerAngles)
+    {
+        Backspin* dev = devices.at(ev.sender);
+        double val[4];
+        euler(-ev.roll, -ev.pitch, ev.yaw, val);
+        OSVR_OrientationState orientationval;
+        memcpy(orientationval.data, val, 4);
+        OSVR_Pose3 pose;
+        pose.rotation = orientationval;
+        //TODO: when 6dof is implemented send translations
+        pose.translation = { 0,0,0 };
```

Comment:
I would comment this out because like Ryan mentioned, Core will take care of it and this is unnecessary